python/its-interqueuemanager/its_iqm/authority/mqtt.py

```diff
@@ -15,73 +15,68 @@
 class Authority:
     def __init__(
         self,
+        instance_id: str,
         cfg: configparser.ConfigParser,
-        update_cb: Callable[[Sequence[Any]], None],
+        update_cb: Callable[[its_iqm.iqm.IQM, dict], None],
     ):
         self.cfg = cfg
         self.update_cb = update_cb
 
-        self.prefix = str()
-        self.suffix = str()
-        # Compare against "not None" instead of truthness, as an
-        # empty prefix is valid and means starting the topic with a /
-        if self.cfg["general"]["prefix"] is not None:
-            self.prefix = f"{self.cfg['general']['prefix']}"
-            # prefix must end with a '/'
-            if self.prefix == "" or self.prefix[-1] != "/":
-                self.prefix += "/"
-        # Test for truthness, because an empty suffix is the same as no suffix
-        if self.cfg["general"]["suffix"]:
-            self.suffix = self.cfg["general"]["suffix"]
-            if self.suffix[-1] != "/":
-                self.suffix += "/"
-
-        self.topic = f"{self.prefix}neighbours/{self.suffix}{self.username}"
+        try:
+            client_id = self.cfg["client_id"]
+        except KeyError:
+            client_id = instance_id
 
         self.authority_client = paho.mqtt.client.Client(
-            client_id=(
-                self.cfg["authority"]["client_id"] or self.cfg["general"]["instance_id"]
-            ),
+            client_id=client_id,
             protocol=paho.mqtt.client.MQTTv5,
         )
         self.authority_client.reconnect_delay_set()
         self.authority_client.username_pw_set(
-            self.cfg["authority"]["username"] or None,
-            self.cfg["authority"]["password"] or None,
+            self.cfg["username"] or None,
```

Comment:
test `self.cfg["username"].str() or None` to well cast to str if it works

Reply:
> test `self.cfg["username"].str() or None` to well cast to str if it works

`self.cfg` is a `configparser.ConfigParser()` instance, so all items **are** string.

Reply:
> [use `.str()`] to well cast to str if it works

After some further thnking about it, I think we should consider that the use
of the configuration should assume the fields are all of the correct type (i.e.
`str` where a strign is expected, `int` where one is expected, and so on...),
and that, if we need to validate that, we should do it when we actualy receive
the configuration, i.e. in the various authority clients; the _file_ and _http_
authority clients should validate and cast the values as appropriate that were
read from the _ini_-style config, and the _mqtt_ authority should check that
the JSON object should match the neighbour description schema (which does not
exist)...